PS4Elo.py

```diff
@@ -0,0 +1,153 @@
+        self.scrub_modifier = scrub_modifier
+
+def getExpectedScore(ranking, other_ranking):
+
+    diff = 10 ** ((other_ranking - ranking) / float(400))
+
+    return float(1) / (1 + diff)
+
+
+def getRankingDelta(ranking, other_ranking, result, k=k_factor):
+
+    if (result == None):
+        return None
+
+    expected_ranking = getExpectedScore(ranking, other_ranking)
+
+    initial_delta = round(k * (result.value - expected_ranking))
+
+    if (initial_delta == 0):
+        if (result == Result.win):
+            return 1
+        if (result == Result.loss):
+            return -1
+
+    return initial_delta
+
+
+def getCombinedRankingForTeam(team, players):
+
+    if len(team) == 0:
+        return 0
+    return sum(map(lambda playerId: getPlayerFromId(players, playerId).ranking, team)) / len(team)
+
+def getOtherTeamRanking(teams, players):
+    merged_teams = reduce(list.__add__, teams)
+    return getCombinedRankingForTeam(merged_teams, players)
+
+def getRankingDeltaForGame(game, players):
+
+    teams = game.teams
+    winning_team_index = game.winning_team_index
+
+    team_rankings = map(lambda team: getCombinedRankingForTeam(team, players), teams)
+
+    players_delta = {}
+    for index, team in enumerate(teams):
+        for playerId in team:
+
+            if (index == winning_team_index):
+                team_result = Result.win
+
+                other_teams = teams[:winning_team_index] + teams[winning_team_index+1 :]
+
+                other_team_ranking = getOtherTeamRanking(other_teams, players)
+            else:
+                team_result = Result.loss
+                other_team_ranking = team_rankings[winning_team_index]
+
+            player_ranking = getPlayerFromId(players, playerId).ranking
+            ranking_delta = getRankingDelta(player_ranking, other_team_ranking, team_result)
+            players_delta[playerId] = ranking_delta
+
+    return players_delta
+
+
+def calculateScrubModifier(player, game):
+    if (player.id in game.scrubs):
+        return scrub_modifier ** game.scrubs[player.id]
+    return 1
+
+def getPlayerFromId(players, playerId):
+    player_found = None
+    if playerId in players:
+        player_found = players[playerId]
+    else:
+        player_found = Player(playerId)
+    return player_found
+
+def calculateRankings(games):
+
+    players = {}
+
+    for game in games:
+        game_players = game.getPlayerIds()
+        for playerId in game_players:
+            if not (playerId in players):
+                players[playerId] = getPlayerFromId(players, playerId)
+
+        individual_ranking_delta = getRankingDeltaForGame(game, players)
+
+        for team in game.teams:
+            for player in map(lambda playerId: getPlayerFromId(players, playerId), team):
+                player.games_played += 1
+
+                scrub_modifier = calculateScrubModifier(player, game)
+
+                ## if the player wins and is sotm they get bonus points TODO fix
```

Comment:
lol

Reply:
lol indeed, i've fixed it to only apply if a player was on the losing team but if you want it the other way, feel free to fly in a PR